subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -708,30 +709,38 @@ u32_t radio_tmr_start(u8_t trx, u32_t ticks_start, u32_t remainder)
 	hal_sw_switch_timer_clear_ppi_config();
 
 #if !defined(CONFIG_BT_CTLR_PHY_CODED) || !defined(CONFIG_SOC_NRF52840)
-	/* NOTE: PPI channel group disable is setup explicitly in sw_switch
-	 *       function when Coded PHY on nRF52840 is supported.
-	 */
-
-	nrf_ppi_channel_endpoint_setup(
-		HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(0),
+	HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_EVT(
```

Comment:
could you call nrf_ppi_event_endpoint_setup here instead ?
This old macros that access the NRF_PPI->CH[chan].EEP / TEP registers will give problems because of not triggering the sideeffects in simulation

Reply:
Actually this patch should do the trick:
```diff
diff --git a/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c b/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c
index 142e4ac956..0ca991dc59 100644
--- a/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c
+++ b/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c
@@ -709,21 +709,17 @@ u32_t radio_tmr_start(u8_t trx, u32_t ticks_start, u32_t remainder)
        hal_sw_switch_timer_clear_ppi_config();
 
 #if !defined(CONFIG_BT_CTLR_PHY_CODED) || !defined(CONFIG_SOC_NRF52840)
-       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_EVT(
-               HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(0)) =
-               HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_EVT(
-                       SW_SWITCH_TIMER_EVTS_COMP(0));
-       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_TASK(
-               HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(0)) =
-               HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_TASK(0);
-
-       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_EVT(
-                       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(1)) =
-               HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_EVT(
-                       SW_SWITCH_TIMER_EVTS_COMP(1));
-       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_TASK(
-               HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(1)) =
-               HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_TASK(1);
+       nrf_ppi_channel_endpoint_setup(
+                       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(0),
+                       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_EVT( SW_SWITCH_TIMER_EVTS_COMP(0)),
+                       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_TASK(0)
+                       );
+
+       nrf_ppi_channel_endpoint_setup(
+                       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(1),
+                       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_EVT( SW_SWITCH_TIMER_EVTS_COMP(1)),
+                       HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_TASK(1)
+                       );
 
        NRF_PPI->CHG[SW_SWITCH_TIMER_TASK_GROUP(0)] =
                HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_0_INCLUDE |
```

Reply:
@aescolar yes, I will fix this. It took some effort to revert the regression commit, but now I can try your suggested code.